registry/txt.go

```diff
@@ -108,7 +108,7 @@ func NewTXTRegistry(provider provider.Provider, txtPrefix, txtSuffix, ownerID st
 }
 
 func getSupportedTypes() []string {
-	return []string{endpoint.RecordTypeA, endpoint.RecordTypeAAAA, endpoint.RecordTypeCNAME, endpoint.RecordTypeNS}
+	return []string{endpoint.RecordTypeA, endpoint.RecordTypeAAAA, endpoint.RecordTypeCNAME, endpoint.RecordTypeNS, endpoint.RecordTypeMX, endpoint.RecordTypeTXT}
```

Comment:
Not sure about the idea to use txt registry for the txt records.
If it's not an integral part of the PR, i would postpone it until we better understand the txt registry roadmap, especially the part of [deprecating old records](https://github.com/kubernetes-sigs/external-dns/pull/5172).

Reply:
using the TXT registry, specifically?